packages/actor-rdf-resolve-quad-pattern-federated/lib/BlankNodeScoped.ts

```diff
@@ -0,0 +1,25 @@
+import * as RDF from "rdf-js";
+
+/**
+ * A blank node that is scoped to a certain source.
+ */
+export class BlankNodeScoped implements RDF.BlankNode {
+  public readonly termType: 'BlankNode' = 'BlankNode';
+  public readonly value: string;
+  /**
+   * This value can be obtained by consumers in query results,
+   * so that this can be passed into another query as an IRI,
+   * in order to obtain more results relating to this (blank) node.
+   */
+  public readonly skolemized: RDF.NamedNode;
+
+  constructor(value: string, skolemized: RDF.NamedNode) {
+    this.value = value;
+    this.skolemized = skolemized;
+  }
+
+  public equals(other: RDF.Term | null | undefined): boolean {
+    return other && other.termType === 'BlankNode' && other.value === this.value;
```

Comment:
strictly `!!other && …`, but that will do

Reply:
Right...
I should really enable those strict null checks in TS, because that would have detected this problem at compile time.